core/src/main/scala/kafka/security/auth/ResourceNameType.scala

```diff
@@ -0,0 +1,49 @@
+/*
```

Comment:
Can't we just use the Java class here rather than duplicating it?

The only reason the other classes are duplicated is that the Scala class came first, and later the Java classes were added for the clients.

Reply:
Sure, I'll clean up in a follow on PR. (It's past midnight here!)

Reply:
Thanks, @big-andy-coates.